### DISCUSSION TITLE
What is the correct formula for calculating Loss function in Yolov5?

### DISCUSSION THREAD

**Discussion by lucysumi** [ultralytics/yolov5#9696] [Q&A]
So, BCE is used for obj and class, while CIOU for regression.
I'm very confused with which is the right way to calculate loss for Yolov5. I read this link and it confused me more.
https://github.com/ultralytics/yolov5/discussions/6276

what exactly is Lbox, Lcls, Lobj  OR  Lclassification, Lconfidence, LCIOU stand for? what does it signify?

Where can I find the correct formula so I can mention it in research paper?

**Comment by glenn-jocher** [comment]
@lucysumi loss is computed in utils/loss.py:
https://github.com/ultralytics/yolov5/blob/master/utils/loss.py
